gto-support-androidx-viewpager2/src/main/java/org/ccci/gto/android/common/androidx/viewpager2/widget/ViewPager2.kt

```diff
@@ -0,0 +1,16 @@
+package org.ccci.gto.android.common.androidx.viewpager2.widget
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager2.widget.ViewPager2
+
+/**
+ * Provides a work-around for https://issuetracker.google.com/issues/181785654
+ */
+fun ViewPager2.whileMaintainingVisibleCurrentItem(block: (RecyclerView.Adapter<*>?) -> Unit) {
+    val adapter = adapter
+    val visible = adapter?.takeIf { it.hasStableIds() }?.getItemId(currentItem)
+    block(adapter)
+    if (visible != null) {
+        (0 until adapter.itemCount).firstOrNull { adapter.getItemId(it) == visible }?.let { setCurrentItem(it, false) }
+    }
```

Comment:
Is the ```0 until adapter.itemCount``` mapping through the pages, and then finding the one that matches ```it``` and comparing that to ```visible``` which is considered the current page? And if they match, then it sets the current item of the ViewPager2 to that page?

Reply:
I should be able to be more clear with this logic by switch to a for loop.

We are basically searching over the adapter to find the index of the item that matches the original `visibleId`, then we set the viewpager to this location.

Reply:
thank you for calling me out of some unnecessary code golf :)